Goldleaf/Source/ns/ns_Service.cpp

```diff
@@ -20,252 +20,128 @@
     {
-        IpcCommand c;
-        ipcInitialize(&c);
-        struct Raw
-        {
-            u64 Magic;
-            u64 CmdId;
-            u64 ApplicationId;
-        } *raw;
-        raw = (struct Raw*)ipcPrepareHeader(&c, sizeof(*raw));
-        raw->Magic = SFCI_MAGIC;
-        raw->CmdId = 27;
-        raw->ApplicationId = ApplicationId;
-        Result rc = serviceIpcDispatch(&nsamsrv);
-        if(R_SUCCEEDED(rc))
-        {
-            IpcParsedCommand r;
-            ipcParse(&r);
-            struct Parsed
-            {
-                u64 Magic;
-                u64 Result;
-            } *resp = (struct Parsed*)r.Raw;
-            rc = resp->Result;
-        }
-        return rc;
+        struct {
+            u64 titleID;
+        } in = { ApplicationId };
+        
+        return serviceDispatchIn(&g_nsAppManSrv, 27, in);
     }
 
     Result CountApplicationContentMeta(u64 ApplicationId, u32 *out_Count)
     {
-        IpcCommand c;
-        ipcInitialize(&c);
-        struct Raw
-        {
-            u64 Magic;
-            u64 CmdId;
-            u64 ApplicationId;
-        } *raw;
-        raw = (struct Raw*)ipcPrepareHeader(&c, sizeof(*raw));
-        raw->Magic = SFCI_MAGIC;
-        raw->CmdId = 600;
-        raw->ApplicationId = ApplicationId;
-        Result rc = serviceIpcDispatch(&nsamsrv);
-        if(R_SUCCEEDED(rc))
-        {
-            IpcParsedCommand r;
-            ipcParse(&r);
-            struct Parsed
-            {
-                u64 Magic;
-                u64 Result;
-                u32 Count;
-            } *resp = (struct Parsed*)r.Raw;
-            rc = resp->Result;
-            if(R_SUCCEEDED(rc)) *out_Count = resp->Count;
-        }
+        struct {
+            u64 titleId;
+        } in = { ApplicationId };
+
+        struct {
+            u32 count;
+        } out;
+
+        Result rc = serviceDispatchInOut(&g_nsAppManSrv, 600, in, out);
+
+        if (R_SUCCEEDED(rc) && out_Count) *out_Count = out.count;
+
         return rc;
     }
 
     Result PushLaunchVersion(u64 ApplicationId, u32 LaunchVersion)
     {
-        IpcCommand c;
-        ipcInitialize(&c);
-        struct Raw
-        {
-            u64 Magic;
-            u64 CmdId;
-            u64 ApplicationId;
-            u32 Version;
-            u32 Pad;
-        } *raw;
-        raw = (struct Raw*)ipcPrepareHeader(&c, sizeof(*raw));
-        raw->Magic = SFCI_MAGIC;
-        raw->CmdId = 36;
-        raw->ApplicationId = ApplicationId;
-        raw->Version = LaunchVersion;
-        raw->Pad = 0;
-        Result rc = serviceIpcDispatch(&nsamsrv);
-        if(R_SUCCEEDED(rc))
-        {
-            IpcParsedCommand r;
-            ipcParse(&r);
-            struct Parsed
-            {
-                u64 Magic;
-                u64 Result;
-            } *resp = (struct Parsed*)r.Raw;
-            rc = resp->Result;
-        }
-        return rc;
+        struct {
+            u64 titleID;

```

Comment:
Application ID (to be more technical), so use sth like appId or appID instead of titleID